src/trifinger_platform_log.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file
+ * @brief Access combined log of a TriFinger platform.
+ * @copyright 2020, Max Planck Gesellschaft.  All rights reserved.
+ * @license BSD 3-clause
+ */
+#include <robot_fingers/trifinger_platform_log.hpp>
+
+namespace robot_fingers
+{
+TriFingerPlatformLog::TriFingerPlatformLog(const std::string &robot_log_file,
+                                           const std::string &camera_log_file)
+    : robot_log_(robot_log_file),
+      camera_log_(camera_log_file),
+      map_robot_to_camera_index_(robot_log_.data.size(), -1)
+{
+    if (robot_log_.data.empty())
+    {
+        throw std::runtime_error("Robot log is empty");
+    }
+    robot_log_start_index_ = robot_log_.data.front().timeindex;
+
+    // for each robot log entry, find the matching camera log entry
+    int i_camera = 0;
+    for (auto i_robot = 0; i_robot < robot_log_.data.size(); i_robot++)
+    {
+        // timestamp in robot log is in seconds, convert to milliseconds
+        auto stamp_robot_ms = robot_log_.data[i_robot].timestamp * 1000;
+
+        while (i_camera < camera_log_.timestamps.size() &&
+               stamp_robot_ms >= camera_log_.timestamps[i_camera])
+        {
+            i_camera++;
+        }
```

Comment:
i think this assumes that there is always a camera observation before the first robot observation. does this make sense?

another related point: you do not check anywhere that the robot timestamp and the camera timestamp are reasonably close. E.g., if I understand correctly, in the case where there is only one camera observation before the robot observations, and one after all the robot observations, it will always return the first.

Reply:
1) There may not be a camera observation before the first robot observation in the log.  In this case, the robot timeindex is mapped to `-1` for the camera which results in an exception when trying to get the camera observation (therefore the `try`-block in the demo script).

2) That's right, but assuming that the data is recorded correctly, that would actually be the correct behaviour.  So I'm not sure how to handle this case otherwise.

Reply:
1. Yes, I see. What I mean is: Is this guaranteed to be satisfied when recording data? Otherwise it will often give exceptions.

2. OK, I guess that's alright then, maybe at most one could do some sanity check on the difference between the timestamps, but maybe that's not straightforward as it depends at what rate the camera was operated. In that case I think it would be fine to leave it as is.

Reply:
1. No it is not guaranteed.  Actually, I think this very likely to not be the case as the camera logger is only started once the first action is received.  This could probably be avoided by adjusting the logger a bit to not start with the current observation but with the oldest in the buffer.  I will take a look.

Reply:
OK, otherwise I guess we could just change this to also allow for the next image, in case the previous one does not exist, but i guess it might be nontrivial to also adjust this accordingly for live operation.

Reply:
I did some analysis of existing log files.  Out of 263 log files, 14 had the issue that there was no camera data for the first few robot steps.  I'm now going to change the camera logger to start logging at the oldest time index instead of the newest one and see if that improves the situation.

Reply:
See https://github.com/open-dynamic-robot-initiative/robot_interfaces/pull/82